app/jobs/regular/publish_topic_to_category.rb

```diff
@@ -12,6 +12,8 @@ def execute(args)
       TopicTimer.transaction do
         TopicPublisher.new(topic, Discourse.system_user, topic_timer.category_id).publish!
       end
+
+      Topic.find(topic.id).inherit_auto_close_from_category
```

Comment:
Any reasons you need to "reload" the topic from the database?

```suggestion
      topic.inherit_auto_close_from_category
```

Reply:
You'll notice that `topic` is taken from the original `TopicTimer`. This ends up storing the topic with its starting `category_id` in `topic`. We need to run `inherit_auto_close_from_category` against the topic _after_ it's published to the new category and has an updated `category_id`. There may be a better way to handle this, but hopefully that at least explains the rationale behind the reload.